dataprofiler/profilers/profile_builder.py

```diff
@@ -1355,11 +1359,132 @@ def _update_row_statistics(self, data, sample_ids=None):
             self.row_has_null_count = len(null_in_row_count)
             self.row_is_null_count = len(null_rows)
 
+    def _get_correlation(self, clean_samples):
+        """
+        Calculate correlation matrix on the cleaned data.
+
+        :param clean_samples: the input cleaned dataset
+        :type clean_samples: dict()
+        """
+        columns = self.options.correlation.columns
+        if columns is None:
+            for col in self._profile:
+                if (self._profile[col].profile['data_type'] not in ['int', 'float']
+                        or self._profile[col].null_count > 0):
+                    clean_samples.pop(col)
+        if len(clean_samples) <= 1:
+            return None
+
+        data = pd.DataFrame(clean_samples)
+        data = data.apply(pd.to_numeric, errors='coerce')
+        return data.corr()
```

Comment:
Not in this PR, but maybe in a subsequent, we could allow user to specify correlation method via options, because I know thats an argument to `.corr()`

Reply:
yeah, we thought about this. Currently only Pearson's correlation is feasible for merging. Spearman or Chi square are tough for now